book/src/dev/rfcs/0004-asynchronous-script-verification.md

```diff
@@ -82,18 +91,157 @@ parallelism.
+- the output is spendable at `height`, to `spend_output_pools`.
+
+We also add a coinbase flag and height to `transparent::Output`s that we look up in
+the state, or get from newly commited blocks:
+```rust
+enum TransparentOutputOrigin {
+    Coinbase {
+        output: transparent::Output,
+        coinbase_height: Height,
+    },
+    NonCoinbase {
+        output: transparent::Output,
+    },
+}
+```
+
+The coinbase flag can be derived from the `OutPoint.index`.
+There is exactly one coinbase transaction per block, and it must have an index of `0`.
+
+The coinbase height can be:
+- looked up from the `OutPoint.hash` in `tx_by_hash`, or
+- retrieved from the newly arrived block.
+
+## Transparent coinbase consensus rules
+[transparent-coinbase-consensus-rules]: #transparent-coinbase-consensus-rules
+
+Specifically, if the UTXO is a transparent coinbase output,
+the service is not required to return a response if:
+- `height` is less than `MIN_TRANSPARENT_COINBASE_MATURITY` after the `OutPoint`'s height, or
+- `spend_pools` is `SomeTransparentOutputs`.
+
+This implements the following consensus rules:
+
+> A transaction MUST NOT spend a transparent output of a coinbase transaction
+> from a block less than 100 blocks prior to the spend.
+>
+> Note that transparent outputs of coinbase transactions include Founders’ Reward
+> outputs and transparent funding stream outputs.
+
+> A transaction with one or more transparent inputs from coinbase transactions
+> MUST have no transparent outputs (i.e.tx_out_count MUST be 0).
+>
+> Inputs from coinbase transactions include Founders’ Reward outputs and funding
+> stream outputs.
+
+https://zips.z.cash/protocol/protocol.pdf#txnencodingandconsensus
+
+## Parallel coinbase checks
+[parallel-coinbase-checks]: #parallel-coinbase-checks
+
+We can perform these coinbase checks asynchronously, in the presence of multiple chain forks,
+as long as the following conditions hold.
+
+We don't mistakenly accept or reject spends to the transparent pool:
+- across all chains, the set of coinbase transaction hashes is disjoint from
+  the set of non-coinbase transaction hashes.
+
+We don't mistakenly accept or reject mature spends:
+- across all chains, the set of coinbase transaction hashes is disjoint from
+  the set of non-coinbase transaction hashes; and
+- across all chains, duplicate coinbase transaction hashes can only occur at
+  exactly the same height.
+
+These properties are implied by the following consensus rules:
+
+Transaction versions 1-4:
+
+> [Pre-Sapling ] If effectiveVersion = 1 or nJoinSplit = 0, then both tx_in_count and tx_out_count MUST be nonzero.
+> ...
+> [Sapling onward] If effectiveVersion < 5, then at least one of tx_in_count, nSpendsSapling, and nJoinSplit MUST be nonzero.
+
+> A coinbase transaction for a block at block height greater than 0
+> MUST have a script that, as its first item, encodes the block height height as follows.
```

Comment:
Minor: not sure if this should be fixed, because it's also like this in the spec, but `height` is repeated here. In the spec, it's because `block height` is in italic (likely representing a number).

Reply:
```suggestion
> MUST have a script that, as its first item, encodes the *block height* height as follows.
```